src/config.rs

```diff
@@ -17,7 +17,7 @@ pub const HOLE_PUNCH_WAIT_FOR_OTHER: bool = true;
 ///
 /// User can opt to provide this in a file, read from it and pass it when required. For optional
 /// fields that are `None`, reasonable defaults will be used.
-#[derive(Debug, Clone, Serialize, Deserialize)]
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
```

Comment:
A natural default doesn't really make sense here as it will be an unusable config (no punchers, no ip-echo servers etc.) and we should not be explicitly giving an API to construct one - maybe create a few minimalistic constructors if you want, which take at the very least
1. `remote_tcp_rendezvous_servers` (e.g., `Config::with_tcp_only(...)`)
2. `remote_udp_rendezvous_servers` (e.g.  `Config::with_udp_only(...)` and in this case create one entry for `udp_hole_punchers` with natural `Default` for `UdpHolePuncher` (so all values will be zero). Then wherever this is used if you get starting-ttl=0 stick to the default OS ttl (i.e., don't do anything with ttl) and if ttl_increment_delay_ms=0 then don't create have a timer started (i.e., no retransmissions). Document this behaviour below in config.rs too.

Ofc the user can always override everything later as all fields are public and put values in a manner where it's an unusable config, but that is a conscious choice at that point.